src/hardware/input/intel8042.cpp

```diff
@@ -0,0 +1,1044 @@
+	//           expecting some incompatible vendor-specific
+	//           extension here) might be unable to recover from.
+	// Various:
+	//     0xb0-0xbd: in general manipulate keyboard controller lines,
+	//                different meanings for different manufacturers
+
+	// Below is the list of controller commands recognized
+	// by this emulator; not all of them are implemented, though
+
+	// Controller memory read/write, most don't have anum values:
+	// 0x00-0x1f: aliases for 0x20-0x3f - obsolete, AMI BIOS
+	// 0x20-0x3f: memory read           - obsolete except for 0x20
+	// 0x40-0x5f: aliases for 0x60-0x7f - obsolete, AMI BIOS
+	// 0x60-0x7f: memory write          - obsolete except for 0x60
+	ReadByteConfig      = 0x20, // usually available
+	WriteByteConfig     = 0x60, // usually available
+	ReadFwCopyright     = 0xa0, // obsolete, some controllers only
+	ReadFwRevision      = 0xa1, // obsolete, some controllers only
+	PasswordCheck       = 0xa4, // obsolete, MCA, some other controllers
+	DisablePortAux      = 0xa7, // usually available
+	EnablePortAux       = 0xa8, // usually available
+	TestPortAux         = 0xa9, // usually available
+	TestController      = 0xaa, // usually available
+	TestPortKbd         = 0xab, // usually available
+	DiagnosticDump      = 0xac, // obsolete, some controllers only
+	DisablePortKbd      = 0xad, // usually available
+	EnablePortKbd       = 0xae, // usually available
+	ReadKbdVersion      = 0xaf, // obsolete, some controllers only
+	ReadInputPort       = 0xc0, // usually available
+	ReadControllerMode  = 0xca, // obsolete, AMI BIOS, VIA
+	WriteControllerMode = 0xcb, // obsolete, AMI BIOS
+	ReadOutputPort      = 0xd0, // usually available
+	WriteOutputPort     = 0xd1, // usually available
+	SimulateInputKbd    = 0xd2, // usually available
+	SimulateInputAux    = 0xd3, // usually available
+	DisableA20          = 0xdd, // obsolete, HP Vectra
+	EnableA20           = 0xdf, // obsolete, HP Vectra
+	ReadTestInputs      = 0xe0, // usually available
+	WriteAux            = 0xd4, // usually available
+	// 0xf0-0xff: pulsing lines, 0xf0 usually available, remaing obsolete
+};
+
+// Byte 0x00 of the controller memory - configuration byte
+
+static union {
+	uint8_t data = 0b0000'0111;
+
+	// bit 0: 1 = byte from keyboard triggers IRQ
+	bit_view<0, 1> is_irq_active_kbd;
+	// bit 1: 1 = byte from aux (mouse) triggers IRQ
+	bit_view<1, 1> is_irq_active_aux;
+	// bit 2: 1 = controller self test passed, 0 = cold boot
+	// bit 3: reserved, should be 0
+
+	// bit 4: 1 = keyboard port disabled
+	bit_view<4, 1> is_disabled_kbd;
+	// bit 5: 1 = aux (mouse) port disabled
+	bit_view<5, 1> is_disabled_aux;
+	// bit 6: 1 = keyboard input should be translated for XT comaptibility
+	bit_view<6, 1> uses_kbd_translation;
+	// bit 7: reserved, should be 0
+
+} config_byte;
+
+static void sanitize_config_byte()
+{
+	config_byte.data = (config_byte.data & 0b0111'0111) | 0b0000'0100;
+}
+
```

Comment:
Maybe include the reserved bits and self-test state, so they can be more self-documenting in the sanitize function?

```c++
static union {
	uint8_t data = 0b0000'0111;

	// bit 0: 1 = byte from keyboard triggers IRQ
	bit_view<0, 1> is_irq_active_kbd;
	// bit 1: 1 = byte from aux (mouse) triggers IRQ
	bit_view<1, 1> is_irq_active_aux;
	// bit 2: 1 = controller self test passed, 0 = cold boot
	bit_view<2, 1> passed_self_test;
	// bit 3: reserved, should be 0
	bit_view<3, 1> reserved_bit_3;
	// bit 4: 1 = keyboard port disabled
	bit_view<4, 1> is_disabled_kbd;
	// bit 5: 1 = aux (mouse) port disabled
	bit_view<5, 1> is_disabled_aux;
	// bit 6: 1 = keyboard input should be translated for XT comaptibility
	bit_view<6, 1> uses_kbd_translation;
	// bit 7: reserved, should be 0
	bit_view<7, 1> reserved_bit_7;
} config_byte;
```

then:

```
static constexpr void sanitize_config_byte()
{
	config_byte.passed_self_test = true;
	config_byte.reserved_bit_3 = 0;
	config_byte.reserved_bit_7 = 0;
}
```

Reply:
OK, changed.